### PR TITLE
Add support for reading, storing and scrobbling MusicBrainz identifiers.

### PR DESCRIPTION
**Pull request by gnu-andrew**
Signed-off-by: Andrew John Hughes gnu_andrew@member.fsf.org

### CODE REVIEW

File: src/xspf/xspf.c [hunk 1]
```diff
@@ -412,7 +413,21 @@ static gboolean xspf_playlist_save (const gchar * filename, VFSFile * file,
...
                     xspf_add_node(track, xs->type, xs->isMeta, xs->xspfName, scratch, scratchi);
             }
+	    scratch = tuple_get_str (tuple, FIELD_MBID);
+	    if (scratch != NULL)
+	    {
+		xmlNodePtr musicbrainz;
+		gchar tmps[94];
+
+		musicbrainz = xmlNewNode(NULL, (xmlChar *) "meta");
+		xmlSetProp(musicbrainz, (xmlChar *) "rel", (xmlChar *) "http://musicbrainz.org/recording");
```
Comment (Freso) [inline src/xspf/xspf.c]: `https://…` please. :)

File: src/xspf/xspf.c [hunk 2]
```diff
@@ -412,7 +413,21 @@ static gboolean xspf_playlist_save (const gchar * filename, VFSFile * file,
...
+		g_snprintf(tmps, sizeof(tmps),
+			   "http://musicbrainz.org/ws/2/recording/%s?inc=artist-credits", scratch);
```
Comment (Freso) [inline src/xspf/xspf.c]: If possible, please use `https://`.

File: src/xspf/xspf.c [hunk 3]
```diff
@@ -412,7 +413,21 @@ static gboolean xspf_playlist_save (const gchar * filename, VFSFile * file,
...
+		xmlAddChild (musicbrainz, xmlNewText ((xmlChar *) tmps));
+		xmlAddChild (track, musicbrainz);
```
Comment (Freso) [inline src/xspf/xspf.c]: What does `track` to here? If it's supposed to use MusicBrainz terminology for the MBID, then it should be `recording` (as also per the two links just above).

**Comment by kaniini** [comment]
I merged the vocabulary parts, leaving the scrobbler alone.  I plan to write a new scrobbler from scratch using QNetwork.

**Pull request closed without merging**